File: code-awesome/java-spring-react/todo/backend/src/pages/create-todo-list.java
import java.util.*;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/todo")
public class CreateTodoList {

    private static final Map<String, String> cuteAnimalNames = new HashMap<>();
    static {
        cuteAnimalNames.put("title", "todoTitle");
        cuteAnimalNames.put("description", "todoDescription");
    }

    @GetMapping("/create-todo-list")
    public String createTodoListPage() {
        // Simulate a form submission
        String title = "New Todo List";
        String description = "This is a new todo list.";

        // Simulate potential content injection (example - dangerous!)
        // NEVER use user-supplied input directly in a database query
        // This is for demonstration purposes only and should be strictly controlled
        // In a real application, use parameterized queries or ORM to prevent SQL injection.

        // Simulating Database interaction
        // In a real application, you would interact with the database here
        // This is a simulated database interaction for demonstration.
        // String sql = "INSERT INTO todo_lists (title, description) VALUES ('" + title + "', '" + description + "')";

        // Simulate error - OWASP Top 10 - Injection (Content)
        // This is a deliberately vulnerable example
        String injectedText = "<script>alert('XSS Attack!')</script>";

        System.out.println("Warning: Potential Content Injection Attempt Detected!");

        return "<h1>Create Todo List</h1>"
                + "<p>Title: " + title + "</p>"
                + "<p>Description: " + description + "</p>"
                + "<p><strong>Note:</strong> This is a demo and vulnerable to content injection.  Do not use this code directly in a production environment.</p>";
    }
}

File: code-awesome/ruby-on-rails-react/todo/backend/src/pages/create-todo-list.rb
# frozen_string_literal: true

class CreateTodoListController < ApplicationController
  skip_before_action :verify_authenticity_token # Disable CSRF protection for demonstration purposes.  DO NOT DO THIS IN PRODUCTION.
  # TODO: Implement robust CSRF protection in a real-world scenario.

  def create
    # Sanitize input to prevent SQL injection (basic example - improve this)
    title = params[:title].strip
    description = params[:description].strip

    # Basic validation (improve this)
    if title.empty? || description.empty?
      render json: { error: "Title and description are required." }, status: :bad_request
      return
    end

    # Database interaction (simulated for demonstration)
    # In a real application, you'd use an ORM like ActiveRecord
    # Example:
    # todo = Todo.new(title: title, description: description)
    # todo.save

    # Simulate database error (demonstrating OWASP top 10 - injection)
    raise "SQL Injection Attempt!" if params[:attack] == "true" # Simulate an attack

    # Successful creation simulation
    render json: { message: "Todo list created successfully!", title: title, description: description }, status: :created
  end
end